source/Engine.cpp

```diff
@@ -2194,7 +2194,11 @@ void Engine::DoCollisions(Projectile &projectile)
 		// For weapons with a trigger radius, check if any detectable object will set it off.
 		double triggerRadius = weapon.TriggerRadius();
 		if(triggerRadius)
-			for(const Body *body : shipCollisions.Circle(projectile.Position(), triggerRadius))
+		{
+			vector<Body *> inRadius;
+			inRadius.reserve(ships.size());
```

Comment:
A trigger radius is typically quite local. What if we use the triggerRadius itself as parameter to reserve? And put a max on it being ships.size(),   so `min(triggerRadius, ships.size())` for the reservation for the vector.

Reply:
```suggestion
			inRadius.reserve(min(triggerRadius, ships.size()));
```